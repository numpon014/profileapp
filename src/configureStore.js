import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';
import createReducer from './reducers';

export default function configureStore(initialState, history) {
  const middlewares = [thunk, routerMiddleware(history)];
  const middlewareEnhancer  = [applyMiddleware(...middlewares)];

  return createStore(
    createReducer(),
    initialState,
    compose(...middlewareEnhancer),
  );
}
