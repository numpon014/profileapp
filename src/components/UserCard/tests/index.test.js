import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { createMemoryHistory } from 'history';
import configureStore from '../../../configureStore';
import UserCard from '../index';
import users from "./users.json";

describe('<UserCard />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render className parameter', () => {
    const className = "user-card"
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <UserCard className={className} user={users[0]}/>
        </ConnectedRouter>
      </Provider>,
    );

    expect(container.querySelector(`div.${className}`)).not.toBeNull();
  });

  it('should render user address information', () => {
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <UserCard user={users[0]}/>
        </ConnectedRouter>
      </Provider>,
    );

    expect(container.querySelectorAll(`div.email span`)[1].textContent).toEqual(users[0].email);
    expect(container.querySelectorAll(`div.phone span`)[1].textContent).toEqual(users[0].phone);
    expect(container.querySelectorAll(`div.website span`)[1].textContent).toEqual(users[0].website);
  });

  it('should render user buttons', () => {
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <UserCard user={users[0]}/>
        </ConnectedRouter>
      </Provider>,
    );

    expect(container.querySelectorAll(`button.like-btn`)).not.toBeNull();
    expect(container.querySelectorAll(`button.edit-btn`)).not.toBeNull();
    expect(container.querySelectorAll(`button.website`)).not.toBeNull();
  });

  it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <UserCard user={users[0]}/>
        </ConnectedRouter>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
