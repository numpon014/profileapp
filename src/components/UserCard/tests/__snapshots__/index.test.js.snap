// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should render a div 1`] = `
<div
  class="sc-dkPtRN fAMsQW"
>
  <div
    class="card"
  >
    <img
      class="card-img-top profile-image"
      src="https://avatars.dicebear.com/v2/avataaars/bret.svg?options[mood][]=happy"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Leanne Graham
      </div>
      <address>
        <div
          class="sc-bdvvtL bHJLNo"
        >
          <div
            class="detail email"
          >
            <span
              class="icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 4a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V4zm2-1a1 1 0 0 0-1 1v.217l7 4.2 7-4.2V4a1 1 0 0 0-1-1H2zm13 2.383-4.758 2.855L15 11.114v-5.73zm-.034 6.878L9.271 8.82 8 9.583 6.728 8.82l-5.694 3.44A1 1 0 0 0 2 13h12a1 1 0 0 0 .966-.739zM1 11.114l4.758-2.876L1 5.383v5.73z"
                />
              </svg>
            </span>
            <span>
              Sincere@april.biz
            </span>
          </div>
        </div>
        <div
          class="sc-bdvvtL bHJLNo"
        >
          <div
            class="detail phone"
          >
            <span
              class="icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.654 1.328a.678.678 0 0 0-1.015-.063L1.605 2.3c-.483.484-.661 1.169-.45 1.77a17.568 17.568 0 0 0 4.168 6.608 17.569 17.569 0 0 0 6.608 4.168c.601.211 1.286.033 1.77-.45l1.034-1.034a.678.678 0 0 0-.063-1.015l-2.307-1.794a.678.678 0 0 0-.58-.122l-2.19.547a1.745 1.745 0 0 1-1.657-.459L5.482 8.062a1.745 1.745 0 0 1-.46-1.657l.548-2.19a.678.678 0 0 0-.122-.58L3.654 1.328zM1.884.511a1.745 1.745 0 0 1 2.612.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z"
                />
              </svg>
            </span>
            <span>
              1-770-736-8031 x56442
            </span>
          </div>
        </div>
        <div
          class="sc-bdvvtL bHJLNo"
        >
          <div
            class="detail website"
          >
            <span
              class="icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm7.5-6.923c-.67.204-1.335.82-1.887 1.855A7.97 7.97 0 0 0 5.145 4H7.5V1.077zM4.09 4a9.267 9.267 0 0 1 .64-1.539 6.7 6.7 0 0 1 .597-.933A7.025 7.025 0 0 0 2.255 4H4.09zm-.582 3.5c.03-.877.138-1.718.312-2.5H1.674a6.958 6.958 0 0 0-.656 2.5h2.49zM4.847 5a12.5 12.5 0 0 0-.338 2.5H7.5V5H4.847zM8.5 5v2.5h2.99a12.495 12.495 0 0 0-.337-2.5H8.5zM4.51 8.5a12.5 12.5 0 0 0 .337 2.5H7.5V8.5H4.51zm3.99 0V11h2.653c.187-.765.306-1.608.338-2.5H8.5zM5.145 12c.138.386.295.744.468 1.068.552 1.035 1.218 1.65 1.887 1.855V12H5.145zm.182 2.472a6.696 6.696 0 0 1-.597-.933A9.268 9.268 0 0 1 4.09 12H2.255a7.024 7.024 0 0 0 3.072 2.472zM3.82 11a13.652 13.652 0 0 1-.312-2.5h-2.49c.062.89.291 1.733.656 2.5H3.82zm6.853 3.472A7.024 7.024 0 0 0 13.745 12H11.91a9.27 9.27 0 0 1-.64 1.539 6.688 6.688 0 0 1-.597.933zM8.5 12v2.923c.67-.204 1.335-.82 1.887-1.855.173-.324.33-.682.468-1.068H8.5zm3.68-1h2.146c.365-.767.594-1.61.656-2.5h-2.49a13.65 13.65 0 0 1-.312 2.5zm2.802-3.5a6.959 6.959 0 0 0-.656-2.5H12.18c.174.782.282 1.623.312 2.5h2.49zM11.27 2.461c.247.464.462.98.64 1.539h1.835a7.024 7.024 0 0 0-3.072-2.472c.218.284.418.598.597.933zM10.855 4a7.966 7.966 0 0 0-.468-1.068C9.835 1.897 9.17 1.282 8.5 1.077V4h2.355z"
                />
              </svg>
            </span>
            <span>
              hildegard.org
            </span>
          </div>
        </div>
      </address>
    </div>
    <div
      class="profile-buttons card-footer"
    >
      <div
        class="p-0 container"
      >
        <div
          class="row"
        >
          <div
            class="button-cols col-sm col-4"
          >
            <div
              class="sc-gsDKAQ cBlxfH"
            >
              <button
                class="like-btn btn btn-link"
                type="button"
              >
                <svg
                  color="red"
                  fill="currentColor"
                  height="18"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: red;"
                  viewBox="0 0 16 16"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="button-cols col-sm col-4"
          >
            <div
              class="sc-gsDKAQ cBlxfH"
            >
              <button
                class="edit-btn btn btn-link"
                type="button"
              >
                <svg
                  color="gray"
                  fill="none"
                  height="18"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  style="color: gray;"
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 20h9"
                  />
                  <path
                    d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="button-cols col-sm col-4"
          >
            <div
              class="sc-gsDKAQ cBlxfH"
            >
              <button
                class="delete-btn btn btn-link"
                type="button"
              >
                <svg
                  color="gray"
                  fill="currentColor"
                  height="18"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: gray;"
                  viewBox="0 0 16 16"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
                  />
                  <path
                    d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
